### PR TITLE
Tests: loading certificate bundles.

### PR DESCRIPTION
**Pull request by jimf5**
Update ssl_verify_client.t to load certificate bundles.

### CODE REVIEW

File: ssl_verify_client.t [hunk 1]
```diff
@@ -53,7 +53,7 @@ http {
         ssl_certificate 1.example.com.crt;
 
         ssl_verify_client on;
-        ssl_client_certificate 2.example.com.crt;
+        ssl_client_certificate 2.example.com.bundle;
```
Comment (pluknet) [inline ssl_verify_client.t:56]: This touches unrelated parts, which will result in poor diagnostics.
I suggest writing a separate server block that specifies ssl_client_certificate with duplicate names.

File: ssl_verify_client.t [hunk 2]
```diff
@@ -128,6 +128,8 @@ foreach my $name ('1.example.com', '2.example.com', '3.example.com') {
 		. "-out $d/$name.crt -keyout $d/$name.key "
 		. ">>$d/openssl.out 2>&1") == 0
 		or die "Can't create certificate for $name: $!\n";
+	system("cat $d/$name.crt $d/$name.crt > $d/$name.bundle") == 0
+		or die "Can't create certificate bundle $name.bundle: $!\n";
```
Comment (pluknet) [inline ssl_verify_client.t:132]: This is not portable, we have read_file / write_file routines for that kind of things.

**Review by pluknet — CHANGES_REQUESTED** [review]
``` diff
commit 33ede378cafce2117de24a076f206174e2b2d6fb (HEAD -> master)
Author: Sergey Kandaurov <pluknet@nginx.com>

    Tests: added ssl_client_certificate test for duplicate CA names.

diff --git a/ssl_verify_client.t b/ssl_verify_client.t
index 56221df..3bbad3a 100644
--- a/ssl_verify_client.t
+++ b/ssl_verify_client.t
@@ -25,7 +25,7 @@ select STDERR; $| = 1;
 select STDOUT; $| = 1;
 
 my $t = Test::Nginx->new()->has(qw/http http_ssl sni socket_ssl_sni/)
-	->has_daemon('openssl')->plan(13);
+	->has_daemon('openssl')->plan(14);
 
 $t->write_file_expand('nginx.conf', <<'EOF');
 
@@ -108,6 +108,17 @@ http {
 
         ssl_verify_client on;
     }
+
+    server {
+        listen       127.0.0.1:8443 ssl;
+        server_name  dup;
+
+        ssl_certificate_key 1.example.com.key;
+        ssl_certificate 1.example.com.crt;
+
+        ssl_verify_client optional;
+        ssl_client_certificate dup.2.example.com.crt;
+    }
 }
 
 EOF
@@ -130,6 +141,8 @@ foreach my $name ('1.example.com', '2.example.com', '3.example.com') {
 		or die "Can't create certificate for $name: $!\n";
 }
 
+$t->write_file('dup.2.example.com.crt', $t->read_file('2.example.com.crt') x 2);
+
 sleep 1 if $^O eq 'MSWin32';
 
 $t->write_file('t', 'SEE-THIS');
@@ -159,6 +172,9 @@ local $TODO = 'broken TLSv1.3 CA list in LibreSSL'
 my $ca = join ' ', get('optional', '3.example.com');
 is($ca, '/CN=2.example.com', 'no trusted sent');
 
+$ca = join ' ', get('dup');
+is($ca, '/CN=2.example.com', 'no duplicates sent');
+
 }
 
 like(get('optional', undef, 'localhost'), qr/421 Misdirected/, 'misdirected');
```

File: ssl_verify_client.t (hunk 1 above)
Comment (jimf5) [reply ssl_verify_client.t:56]: I agree that a separate server block is better.

File: ssl_verify_client.t (hunk 2 above)
Comment (jimf5) [reply ssl_verify_client.t:132]: Yes, that's right.

**Comment by jimf5** [comment]
I have updated this PR with your patch.

**Review by pluknet — APPROVED** [review]

**Pull request merged**